main/exercise/ptests_category.php

```diff
@@ -0,0 +1,255 @@
+
+$htmlHeadXtra[] = '
+<script>
+	function confirmDelete(in_txt, in_id) {
+		var oldbgcolor = document.getElementById(in_id).style.backgroundColor;
+		document.getElementById(in_id).style.backgroundColor="#AAFFB0";
+		if (confirm(in_txt)) {
+			return true;
+		} else {
+			document.getElementById(in_id).style.backgroundColor = oldbgcolor;
+			return false;
+		}
+	}
+</script>';
+
+$nameTools = '';
+
+require_once __DIR__.'/../inc/global.inc.php';
+
+$this_section = SECTION_COURSES;
+
+api_protect_course_script(true);
+
+if (!api_is_allowed_to_edit()) {
+    api_not_allowed(true);
+}
+
+$category = new PTestCategory();
+$courseId = api_get_course_int_id();
+$sessionId = api_get_session_id();
+$exerciseId = (int) $_GET['exerciseId'];
+
+// get from session
+$objExercise = Session::read('objExercise');
+
+// Exercise object creation.
+if (!is_object($objExercise) || $objExercise->selectPtType() != EXERCISE_PT_TYPE_PTEST) {
+    // construction of the Exercise object
+    $objExercise = new Exercise();
+
+    // creation of a new exercise if wrong or not specified exercise ID
+    if ($exerciseId) {
+        $parseQuestionList = $showPagination > 0 ? false : true;
+        if ($editQuestion) {
+            $parseQuestionList = false;
+            $showPagination = true;
+        }
+        $objExercise->read($exerciseId, $parseQuestionList);
+    }
+    // saves the object into the session
+    Session::write('objExercise', $objExercise);
+}
+
+// Exercise can be edited in their course.
+if ($objExercise->sessionId != $sessionId) {
+    api_not_allowed(true);
+}
+
+// breadcrumbs
+$interbreadcrumb[] = [
+    "url" => "ptest_admin.php?exercise_id=".$objExercise->selectId()."&".api_get_cidreq(),
+    "name" => get_lang('Exercises'),
+];
+
+$action = isset($_GET['action']) ? $_GET['action'] : '';
+$content = '';
+
+switch ($action) {
+    case 'addcategory':
+        $content = add_category_form('addcategory');
+        break;
+    case 'editcategory':
+        $content = edit_category_form('editcategory');
+        break;
+    case 'deletecategory':
+        delete_category_form();
+        break;
+}
+
+Display::display_header(get_lang('Category'));
+displayActionBar();
+echo $content;
+echo $category->displayCategories($exerciseId, $courseId, $sessionId);
+Display::display_footer();
+
+/**
+ * Form to edit a category.
+ *
+ * @todo move to PTestCategory.class.php
+ *
+ * @param string $action
+ */
+function edit_category_form($action)
+{
+    $exerciseId = (int) $_GET['exerciseId'];
+    $action = Security::remove_XSS($action);
+    if (isset($_GET['category_id']) && is_numeric($_GET['category_id'])) {
+        $category_id = intval($_GET['category_id']);
+        $objcat = new PTestCategory();
+        $objcat = $objcat->getCategory($category_id);
+        $form = new FormValidator(
+            'note',
+            'post',
+            api_get_self().'?exerciseId='.$exerciseId.'&action='.$action.'&category_id='.$category_id.'&'.api_get_cidreq()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line exceeds maximum limit of 120 characters; contains 122 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5018478)